agent/consul/leader_connect_ca.go

```diff
@@ -550,13 +563,10 @@ func (c *CAManager) primaryInitialize(provider ca.Provider, conf *structs.CAConf
 		rootCA.IntermediateCerts = activeRoot.IntermediateCerts
 		c.setCAProvider(provider, rootCA)
 
+		c.logger.Info("initialized primary datacenter CA from existing CARoot with provider", "provider", conf.Provider)
```

Comment:
This is unrelated, but something I noticed while working on this.

I'm pretty sure that previously we would not log anything when a server initializes itself using an existing stored `CARoot`. I guess this would happen if the raft leader election on any cluster that already had a root setup. 

I made the log message slightly different, so that it's clear this initialization was from an existing `CARoot`.